### PR TITLE
Docs: Try to improve the docs about custom CAs

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Documentation

### Description

After some experience with multiple users try to get custom CAs working, I prepared an minor update to the relevant section of the documentation which important things which are missing or not obvious:

* Everything should be prepared before cluster deployment
* Handling for Intermediate CAs and CA chains needs to be explained
* The OpenSSL command is IMHO not needed. This is the same what CO does automatically. So why would we tell users to use that.
* Move the `kubectl label` and `oc label` commands to separate step. Many users including me missed them because they were hidden at the end of a very long command. This should hopefully make it a bit more noticeable

Further feedback would be welcome.

### Checklist

- [x] Update documentation

### CODE REVIEW

File: documentation/book/proc-installing-your-own-ca-certificates.adoc [hunk 1]
```diff
@@ -10,12 +10,17 @@ This procedure describes how to install your own CA certificates and private key
...
-openssl req -x509 -new -days _<validity>_ --nodes -out ca.crt -keyout ca.key
-
+** If you want to use cluster or clients CA which is not a Root CA, you have to include the whole chain in the certificate file.
+The chain should be in the following order:
++
+1. The cluster or clients CA
+2. One or more intermediate CAs
+3. The root CA
++
+** All CAs in the chain should be marked as CA in the Basic Constraints extension.
```
Comment (ppatierno) [inline documentation/book/proc-installing-your-own-ca-certificates.adoc]: As far as I understand, the basic constraints extension is a list of constraints so we could say.
I would change "should be marked ... " in "should have CA ... " to be clear that they could have something more for other purposes but they need CA for Strimzi. Using "marked" sounds to me that they have to have just CA in the extension.
Maybe it's just me :-)

Comment (scholzj) [reply documentation/book/proc-installing-your-own-ca-certificates.adoc]: I rephrased it a bit. I hope it works better for you now.

File: documentation/book/proc-installing-your-own-ca-certificates.adoc [hunk 2]
```diff
@@ -10,12 +10,17 @@ This procedure describes how to install your own CA certificates and private key
 .Prerequisites
 
 * The Cluster Operator is running.
-* A `Kafka` resource within {ProductPlatformName}
-* Your own X.509 certificates and keys in PEM format for the cluster CA or clients CA. For example, these could be generated by `openssl`, using a command such as:
+* A Kafka cluster is not yet deployed
```
Comment (ncbaratta) [inline documentation/book/proc-installing-your-own-ca-certificates.adoc]:
```suggestion
* A Kafka cluster is not yet deployed.
```

File: documentation/book/proc-installing-your-own-ca-certificates.adoc [hunk 3]
```diff
@@ -10,12 +10,17 @@ This procedure describes how to install your own CA certificates and private key
...
+* Your own X.509 certificates and keys in PEM format for the cluster CA or clients CA.
 +
-[source,shell,subs="+quotes"]
-openssl req -x509 -new -days _<validity>_ --nodes -out ca.crt -keyout ca.key
-
+** If you want to use cluster or clients CA which is not a Root CA, you have to include the whole chain in the certificate file.
```
Comment (ncbaratta) [inline documentation/book/proc-installing-your-own-ca-certificates.adoc]:
```suggestion
** If you want to use a cluster or clients CA which is not a Root CA, you have to include the whole chain in the certificate file.
```

**Pull request merged**